### PR TITLE
docs(datasync): create datasync blog post

### PR DESCRIPTION
**Pull request by ssd71**

### CODE REVIEW

File: website/blog/2020-10-1-graphback-datasync.md [hunk 1]
```diff
@@ -0,0 +1,147 @@
+---
+title: Rising above GraphQL subscriptions with Graphback DataSync
+author: Shibani Shankar Dash
+author_title: GSoC '20 Student @ Aerogear
```
Comment (wtrocki) [inline website/blog/2020-10-1-graphback-datasync.md:4]: This sounds better IMHO
```suggestion
author_title: Aerogear community contributor
```

File: website/blog/2020-10-1-graphback-datasync.md [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+author_url: https://github.com/ssd71
+author_image_url: https://avatars1.githubusercontent.com/u/20183094?s=460&u=dc7003f2886fbefe348ab88db9db6087fd23cb29&v=4
```
Comment (wtrocki) [inline website/blog/2020-10-1-graphback-datasync.md:6]: Really ? :)

**Comment by wtrocki** [comment]
Great post. I would drop something fun - like meme or something because it looks like page from `87 Engineering Journal` rather than modern software solution

**Comment by wtrocki** [comment]
Photo for reference https://ieeexplore.ieee.org/document/553183

File: website/blog/2020-10-1-graphback-datasync.md [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+title: Rising above GraphQL subscriptions with Graphback DataSync
+author: Shibani Shankar Dash
+author_title: Aerogear community contributor
+author_url: https://github.com/ssd71
+author_image_url: https://avatars1.githubusercontent.com/u/20183094?s=460&u=dc7003f2886fbefe348ab88db9db6087fd23cb29&v=4
+tags: [graphql, graphback, graphqlcrud, graphql-js, nodejs, datasync, graphql-subscriptions]
+image: https://graphback.dev/img/logo.png
+---
+
+GraphQL subscriptions are a perfect fit for situations where users like to receive instant updates for some specific types of data. Subscriptions are very flexible and allow developers to connect directly to the stream of the changes generated on the server side.
```
Comment (craicoverflow) [inline website/blog/2020-10-1-graphback-datasync.md:11]:
```suggestion
GraphQL Subscriptions are a perfect fit for situations where users like to receive instant updates for some specific types of data. Subscriptions are very flexible and allow developers to connect directly to the stream of the changes generated on the server side.
```

File: website/blog/2020-10-1-graphback-datasync.md [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+## Delta queries
+Delta queries are a concept that allow users to perform a query that will return partial data based on a specific separator that is usually represented as a sequence or timestamp or any date series that will separate a sorted dataset into data we have seen and a delta that should be returned to clients.
+
+![](../static/img/syncdiagram.png)
+
+## Delete is challenging
+Not every dataset can support delta queries. Every deletion of the data will remove it from the database which means that sorting and splitting the dataset to retrieve updates will destroy it.
+
+## Conflict resolution
+In an offline-first application, where a client may often switch between being online and offline, conflicts between client-side and server-side data are almost guaranteed to occur. It is of utmost importance these conflicts are detected and some mechanism is  in place to resolve these conflicts.
```
Comment (craicoverflow) [inline website/blog/2020-10-1-graphback-datasync.md:22]:
```suggestion
In an offline-first application, where a client may often switch between being online and offline, conflicts between client-side and server-side data are almost guaranteed to occur. It is of utmost importance these conflicts are detected and some mechanism is in place to resolve these conflicts.
```

File: website/blog/2020-10-1-graphback-datasync.md [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+
+- Stop deleting data in the table to introduce soft deletes 
+- Have separate table for delete operations and merging results when sending data to clients
+- Have separate table that can be used to query diffs 
+
+These concepts are very common and are often needed when building applications that will consume large amount of data that needs to be available on the client - especially when the client is designed to work without network connectivity.
+
+### GraphQL CRUD spec for delta queries
+
+That is why we have built and included delta operations as well as conflict resolution into GraphQLCRUD.org specification.
```
Comment (craicoverflow) [inline website/blog/2020-10-1-graphback-datasync.md:38]: Hyperlink to GraphQLCRUD

File: website/blog/2020-10-1-graphback-datasync.md [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+Today we are glad to announce that DeltaQueries as well as Server-side Conflict Resolution are officially available as part of the Graphback.dev project that implements this specification.
+Graphback is a library that lets you auto-generate your GraphQL Data layer at runtime. The latest release of Graphback provides support for delta queries as well as server-side conflict resolution.
+
+## What is DataSync?
+
+DataSync is short for data synchronization and that’s what we set out to do with this package. When a GraphQL client goes offline for sometime, e.g. because of a poor network connection, the data cached by the client can get outdated pretty fast and without a live subscription to fetch those changes it becomes imperative that a client provide some way to be up to date with the data as soon as it comes back online. That is DataSync.
```
Comment (craicoverflow) [inline website/blog/2020-10-1-graphback-datasync.md:44]: I see "this package" mentioned but no link or reference to `@graphback/datasync`.

File: website/blog/2020-10-1-graphback-datasync.md [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+        }
+      ],
+      "lastSync": "1591852700920",
+      "limit": null
+    }
+  }
+}
+```
+
+Notice that we not only get the required fields like id and text, but also `_lastUpdatedAt` as well as if the post was `_deleted` , all of which are automatically maintained by DataSync.
```
Comment (craicoverflow) [inline website/blog/2020-10-1-graphback-datasync.md:85]:
```suggestion
Notice that we not only get the required fields like id and text, but also `_lastUpdatedAt` as well as if the post was `_deleted`, all of which are automatically maintained by DataSync.
```

File: website/blog/2020-10-1-graphback-datasync.md [hunk 8]
```diff
@@ -0,0 +1,148 @@
...
+
+scalar GraphbackObjectID
+```
+This is going to be our data model. Notice that both types are annotated with @model and @datasync. The former is required for Graphback to recognise model types and the latter is required to enable DataSync for a specific model.
+
+Visit the [docs](https://graphback.dev/docs/next/datasync/intro) for a complete reference to all Graphback supported annotations as well as a more involved introduction to DataSync.
+
+Edit the data model as desired and move on.
+
+- `npx graphql-serve serve model.graphql --datasync -p 4000`
```
Comment (craicoverflow) [inline website/blog/2020-10-1-graphback-datasync.md:145]:
```suggestion
```shell
npx graphql-serve serve model.graphql --datasync -p 4000
```
```

File: website/blog/2020-10-1-graphback-datasync.md [hunk 9]
```diff
@@ -0,0 +1,148 @@
+---
+title: Rising above GraphQL subscriptions with Graphback DataSync
```
Comment (craicoverflow) [inline website/blog/2020-10-1-graphback-datasync.md:2]:
```suggestion
title: Rising above GraphQL Subscriptions with Graphback DataSync
```

**Comment by wtrocki** [comment]
@craicoverflow Amazing review! I think we can improve it and merge on behalf of @ssd71

**Comment by machi1990** [comment]
All good, we are only missing a TL;DR section as:
https://github.com/aerogear/graphback/blob/master/website/blog/2020-07-22-new-graphback-release-0-14.md#tldr which is truncated with the `<!--truncate-->`

This was the whole blog post will not appear in the blog landing page but a summary of it: see https://graphback.dev/blog/

**Review by machi1990 — APPROVED** [review]
I have pushed a commit based on reviews from @craicoverflow and my own review. I think this is good enough to go for now. We can improve later on.

**Comment by machi1990** [comment]
Merged. Thanks @ssd71 for the good work as usual.

**Pull request merged**